app/src/main/java/com/mobile/tandil/javabaseproject/mvp/presenter/ReservationPresenter.java

```diff
@@ -0,0 +1,77 @@
+package com.mobile.tandil.javabaseproject.mvp.presenter;
+
+import com.mobile.tandil.javabaseproject.R;
+import com.mobile.tandil.javabaseproject.entity.Reservation;
+import com.mobile.tandil.javabaseproject.enums.ValidationEnums;
+import com.mobile.tandil.javabaseproject.listener.ListenerPickers;
+import com.mobile.tandil.javabaseproject.mvp.contract.ReservationContract;
+
+import java.text.Format;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+public class ReservationPresenter implements ReservationContract.Presenter {
+
+    private final ReservationContract.View view;
+    private final ReservationContract.Model model;
+
+    public ReservationPresenter(ReservationContract.View view, ReservationContract.Model model) {
+        this.view = view;
+        this.model = model;
+    }
+
+    @Override
+    public void showPickers(ListenerPickers listenerPickers, Boolean startOrEnd) {
+        view.showDateAndTime(listenerPickers, startOrEnd);
+    }
+
+    public void listenStartingDate(Calendar startingDate) {
+        view.showStartingDate(formatAndShow(startingDate));
+    }
+
+    public void listenEndingDate(Calendar endingDate) {
+        view.showEndingDate(formatAndShow(endingDate));
+    }
+
+    private String formatAndShow(Calendar date) {
```

Comment:
please extract this into a `DateUtils` class

Reply:
done